aeneas/src/x86/X86MacroAssembler.v3

```diff
@@ -51,22 +73,51 @@ class X86MacroAssembler extends X86Assembler {
 		push(s);            // push value from source memory onto the stack
 		pop(X86Addr.!(d));  // pop value off stack into destination memory
 	}
+	// a macro to move between any two SSE register/memory operands
+	def movsd_sm_sm(d: SSERm, s: SSERm, scratch: SSEReg) {
+		if (s == d) return;
+		if (SSEReg.?(d)) return movsd_s_sm(SSEReg.!(d), s);
+		if (SSEReg.?(s)) return movsd_sm_s(d, SSEReg.!(s));
+		if (scratch != null) {
+			movsd_s_sm(scratch, s);
+			movsd_sm_s(d, scratch);
+		}
+		// push(s);            // push value from source memory onto the stack
```

Comment:
Do we still need the case where no scratch register is available?

Reply:
I don't imagine so. X86CodeGen currently explicitly supplies `xmm7` as the scratch register. I'll remove the comments.